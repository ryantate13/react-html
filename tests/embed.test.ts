import {embed} from '../src';

describe('embed', () => {

    it('renders with empty attributes', () => {
        expect(embed({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(embed()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(embed({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(embed({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(embed({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(embed({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(embed({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(embed({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(embed({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(embed({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(embed({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(embed({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(embed({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(embed({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(embed({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(embed({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(embed({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(embed({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(embed({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(embed({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(embed({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(embed({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(embed({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(embed({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(embed({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(embed({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(embed({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(embed({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(embed({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(embed({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(embed({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(embed({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(embed({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(embed({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(embed({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(embed({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(embed({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(embed({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(embed({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(embed({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(embed({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(embed({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(embed({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(embed({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(embed({translate: "no"})).toMatchSnapshot();
    });

    it('renders with height="1"', () => {
        expect(embed({height: 1})).toMatchSnapshot();
    });

    it('renders with src="https://google.com/"', () => {
        expect(embed({src: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with type="application/json"', () => {
        expect(embed({type: "application/json"})).toMatchSnapshot();
    });

    it('renders with width="1"', () => {
        expect(embed({width: 1})).toMatchSnapshot();
    });
});