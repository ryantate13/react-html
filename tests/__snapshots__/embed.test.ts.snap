// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`embed renders with accessKey="a,b,c,d" 1`] = `
<embed
  accessKey="a,b,c,d"
/>
`;

exports[`embed renders with autoCapitalize="characters" 1`] = `
<embed
  autoCapitalize="characters"
/>
`;

exports[`embed renders with autoCapitalize="none" 1`] = `
<embed
  autoCapitalize="none"
/>
`;

exports[`embed renders with autoCapitalize="off" 1`] = `
<embed
  autoCapitalize="off"
/>
`;

exports[`embed renders with autoCapitalize="on" 1`] = `
<embed
  autoCapitalize="on"
/>
`;

exports[`embed renders with autoCapitalize="sentences" 1`] = `
<embed
  autoCapitalize="sentences"
/>
`;

exports[`embed renders with autoCapitalize="words" 1`] = `
<embed
  autoCapitalize="words"
/>
`;

exports[`embed renders with className="foo bar baz" 1`] = `
<embed
  className="foo bar baz"
/>
`;

exports[`embed renders with contentEditable="false" 1`] = `
<embed
  contentEditable="false"
/>
`;

exports[`embed renders with contentEditable="true" 1`] = `
<embed
  contentEditable="true"
/>
`;

exports[`embed renders with dir="auto" 1`] = `
<embed
  dir="auto"
/>
`;

exports[`embed renders with dir="ltr" 1`] = `
<embed
  dir="ltr"
/>
`;

exports[`embed renders with dir="rtl" 1`] = `
<embed
  dir="rtl"
/>
`;

exports[`embed renders with draggable="false" 1`] = `
<embed
  draggable="false"
/>
`;

exports[`embed renders with draggable="true" 1`] = `
<embed
  draggable="true"
/>
`;

exports[`embed renders with empty attributes 1`] = `<embed />`;

exports[`embed renders with height=1 1`] = `
<embed
  height={1}
/>
`;

exports[`embed renders with hidden=false 1`] = `
<embed
  hidden={false}
/>
`;

exports[`embed renders with hidden=true 1`] = `
<embed
  hidden={true}
/>
`;

exports[`embed renders with id="TEST" 1`] = `
<embed
  id="TEST"
/>
`;

exports[`embed renders with inputMode="decimal" 1`] = `
<embed
  inputMode="decimal"
/>
`;

exports[`embed renders with inputMode="email" 1`] = `
<embed
  inputMode="email"
/>
`;

exports[`embed renders with inputMode="none" 1`] = `
<embed
  inputMode="none"
/>
`;

exports[`embed renders with inputMode="numeric" 1`] = `
<embed
  inputMode="numeric"
/>
`;

exports[`embed renders with inputMode="search" 1`] = `
<embed
  inputMode="search"
/>
`;

exports[`embed renders with inputMode="tel" 1`] = `
<embed
  inputMode="tel"
/>
`;

exports[`embed renders with inputMode="text" 1`] = `
<embed
  inputMode="text"
/>
`;

exports[`embed renders with inputMode="url" 1`] = `
<embed
  inputMode="url"
/>
`;

exports[`embed renders with is="foo" 1`] = `
<embed
  is="foo"
/>
`;

exports[`embed renders with itemID="https://google.com/" 1`] = `
<embed
  itemID="https://google.com/"
/>
`;

exports[`embed renders with itemProp="https://google.com/" 1`] = `
<embed
  itemProp="https://google.com/"
/>
`;

exports[`embed renders with itemRef="foo bar" 1`] = `
<embed
  itemRef="foo bar"
/>
`;

exports[`embed renders with itemScope=false 1`] = `
<embed
  itemScope={false}
/>
`;

exports[`embed renders with itemScope=true 1`] = `
<embed
  itemScope={true}
/>
`;

exports[`embed renders with itemType="https://google.com/" 1`] = `
<embed
  itemType="https://google.com/"
/>
`;

exports[`embed renders with lang="" 1`] = `
<embed
  lang=""
/>
`;

exports[`embed renders with lang="es-419" 1`] = `
<embed
  lang="es-419"
/>
`;

exports[`embed renders with slot="TEST" 1`] = `
<embed
  slot="TEST"
/>
`;

exports[`embed renders with spellCheck="false" 1`] = `
<embed
  spellCheck="false"
/>
`;

exports[`embed renders with spellCheck="true" 1`] = `
<embed
  spellCheck="true"
/>
`;

exports[`embed renders with src="https://google.com/" 1`] = `
<embed
  src="https://google.com/"
/>
`;

exports[`embed renders with style={"color":"#000000"} 1`] = `
<embed
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`embed renders with tabIndex=1 1`] = `
<embed
  tabIndex={1}
/>
`;

exports[`embed renders with title="TEST" 1`] = `
<embed
  title="TEST"
/>
`;

exports[`embed renders with translate="no" 1`] = `
<embed
  translate="no"
/>
`;

exports[`embed renders with translate="yes" 1`] = `
<embed
  translate="yes"
/>
`;

exports[`embed renders with type="application/json" 1`] = `
<embed
  type="application/json"
/>
`;

exports[`embed renders with width=1 1`] = `
<embed
  width={1}
/>
`;

exports[`embed renders without attributes 1`] = `<embed />`;
